jit.js

```diff
@@ -692,7 +932,12 @@ to single-step.
     generateLabel: function() {
         // remember label position for deleteUnneededLabels()
         this.sourceLabels[this.prevPC] = this.source.length;
-        this.source.push("case ", this.prevPC, ":\n");
+        if (!this.suppressNextLabel) {
+            this.source.push("case ", this.prevPC, ":\n");
+        } else {
+            this.source.push("\n");
+        }
+        this.suppressNextLabel = false;
```

Comment:
Oh, it's very simple: all labels that were not 'enabled' in `needsLabel` are deleted at the end. Just look at `deleteUnneededLabels()`. So all you need to do is not set `needsLabel[pc]` to `true`.